### PR TITLE
Support Node 4.6 LTS

### LINKED ISSUE #29: Unexpected Token: {
Cannot attach version `1.2.0-beta11` to Babel. As soon as I do, the following stack is thrown:

```
ERROR in ./src/index.js
Module build failed: SyntaxError: Unexpected token {
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:404:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at ~/node_modules/babel-core/lib/transformation/file/options/option-manager.js:174:20
    at Array.map (native)
    at Function.normalisePlugins (~/node_modules/babel-core/lib/transformation/file/options/option-manager.js:154:20)
    at OptionManager.mergeOptions (~/node_modules/babel-core/lib/transformation/file/options/option-manager.js:229:36)
    at OptionManager.init (~/node_modules/babel-core/lib/transformation/file/options/option-manager.js:374:12)
    at File.initOptions (~/node_modules/babel-core/lib/transformation/file/index.js:216:65)
    at new File (~/node_modules/babel-core/lib/transformation/file/index.js:139:24)
    at Pipeline.transform (~/node_modules/babel-core/lib/transformation/pipeline.js:46:16)
    at transpile (~/node_modules/babel-loader/lib/index.js:41:20)
    at Object.module.exports (~/node_modules/babel-loader/lib/index.js:138:12)
```

It's nothing to do with my source file: `src/index.js`. To verify, I set its contents to:

```js
console.log('nearly empty');
```

and the error is still thrown.

Reverting to `1.0.0-legacy` allowed me to continue working.

### PR DESCRIPTION
**Pull request by lukeed**
* Simplify a few helper functions & `if`/`elif` expressions.

* Extract `vNodeFlags` definitions to own file for easier reading.

* Revert object destructuring ("re-structuring"? lol) for `4.6` support. 
  * tested down to `4.2`

* Closes #29 

---

**Next?**

- There are a few helper files that are never required.

- The `attrsSvg` file is also never used.

- Add tests

- Attach Travis (or similar)

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -2,31 +2,10 @@
...
-
-	SvgElement: 1 << 7,
-	MediaElement: 1 << 8,
-	InputElement: 1 << 9,
-	TextareaElement: 1 << 10,
-	SelectElement: 1 << 11,
-	Fragment: 1 << 12,
-	Void: 1 << 13
-};
+const VNodeFlags = require('./flags');
```
Comment (LukeSheard) [inline lib/index.js]: Instead of doing this we should just add `inferno-vnode-flags` as a dependency and require that package.

Comment (lukeed) [inline lib/index.js]: Not a bad idea, especially if used elsewhere.

File: lib/index.js [hunk 2]
```diff
@@ -71,9 +50,9 @@ function getVNodeType(t, type) {
 		flags = VNodeFlags.ComponentUnknown;
 	}
 	return {
-		type,
+		type: type,
 		isComponent: component,
```
Comment (LukeSheard) [inline lib/index.js]: To keep this consistent we should rename `component` to `isComponent` for readability.

Comment (lukeed) [inline lib/index.js]: Oh, that's flat-out wrong. Thanks!

**Comment by Havunen** [comment]
LGTM

File: lib/index.js (hunk 2 above)
Comment (LukeSheard) [inline lib/index.js]: Hmm. That's annoying. Ah well.

**Comment by lukeed** [comment]
@Havunen Thanks! Thoughts on my next steps? Happy to go forward & take care of them.

**Pull request merged**